elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import java.net.URISyntaxException;
+
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.ElideResponse;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.async.models.QueryType;
+import com.yahoo.elide.graphql.QueryRunner;
+import com.yahoo.elide.security.User;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Runnable thread for executing the query provided in Async Query.
+ * It will also update the query status and result object at different
+ * stages of execution.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class AsyncQueryThread implements Runnable {
+
+    private AsyncQuery queryObj;
+    private User user;
+    private Elide elide;
+    private QueryRunner runner;
+    private AsyncQueryDAO asyncQueryDao;
+
+    @Override
+    public void run() {
+        processQuery();
+    }
+
+    /**
+     * This is the main method which processes the Async Query request, executes the query and updates
+     * values for AsyncQuery and AsyncQueryResult models accordingly.
+     */
+    protected void processQuery() {
+        try {
+            // Change async query to processing
+            asyncQueryDao.updateAsyncQuery(queryObj.getId(), (asyncQuery) -> {
+                asyncQuery.setStatus(QueryStatus.PROCESSING);
+                });
+            ElideResponse response = null;
+            log.debug("AsyncQuery Object from request: {}", queryObj);
+            if (queryObj.getQueryType().equals(QueryType.JSONAPI_V1_0)) {
+                MultivaluedMap<String, String> queryParams = getQueryParams(queryObj.getQuery());
+                log.debug("Extracted QueryParams from AsyncQuery Object: {}", queryParams);
+                response = elide.get(getPath(queryObj.getQuery()), queryParams, user);
+                log.debug("JSONAPI_V1_0 getResponseCode: {}, JSONAPI_V1_0 getBody: {}", response.getResponseCode(), response.getBody());
+            }
+            else if (queryObj.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
+                response = runner.run(queryObj.getQuery(), user);
+                log.debug("GRAPHQL_V1_0 getResponseCode: {}, GRAPHQL_V1_0 getBody: {}", response.getResponseCode(), response.getBody());
+            }
+            if (response != null){
```

Comment:
We are logging the response above before we check for null here.

Reply:
Technically, I don't think this ever returns null.  I think it either returns non-null or throws an exception.

Reply:
Instead of checking for null, we can just add an assert here that will throw an exception if response is null.